### PR TITLE
Periodically verify that health checks are still being executed

### PR DESCRIPTION
**Pull request by dump247**
An Error (e.g. out of memory) can prevent a health check from being
rescheduled to run again. This prevents the end point from ever
being marked healthy again.

### CODE REVIEW

**Comment by dump247** [comment]
@ohhatiya i added a test

File: core/src/test/java/com/bazaarvoice/ostrich/pool/ServicePoolTest.java [hunk 1]
```diff
@@ -405,6 +409,49 @@ public Void call(Service service) throws ServiceException {
     }
 
     @Test
+    public void testHealthCheckIsRescheduled() {
+        ArgumentCaptor<ServicePool.HealthCheckVerifier> verifyCapture = ArgumentCaptor.forClass(ServicePool.HealthCheckVerifier.class);
+        verify(_healthCheckExecutor).scheduleAtFixedRate(verifyCapture.capture(), anyLong(), anyLong(), any(TimeUnit.class));
+        ServicePool.HealthCheckVerifier verify = verifyCapture.getValue();
```
Comment (ohhatiya) [inline core/src/test/java/com/bazaarvoice/ostrich/pool/ServicePoolTest.java]: This name is a little bit confusing when mixed with calls to Mockito's verify() methods.

**Comment by dump247** [comment]
@ohhatiya variable renamed

**Comment by ohhatiya** [comment]
:metal:

**Pull request merged**